src/js/player.js

```diff
@@ -451,6 +448,12 @@ vjs.Player.prototype.onLoadedAllData;
 vjs.Player.prototype.onPlay = function(){
   this.removeClass('vjs-paused');
   this.addClass('vjs-playing');
+
+  // hide the poster when the user hits play
+  // https://html.spec.whatwg.org/multipage/embedded-content.html#dom-media-play
+  if (!this.hasStarted()) {
```

Comment:
The hasStarted method already [handles this check](https://github.com/videojs/video.js/blob/d5bbd80db989ac7a96abfafd461591d32cc44076/src/js/player.js#L413), if you want to simplify this.